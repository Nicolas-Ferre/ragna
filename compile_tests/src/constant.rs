fn main() {}

#[ragna::gpu]
mod gpu {
    const CONSTANT: i32 = 30;

    fn run() {
        CONSTANT = 20;
    }
}
