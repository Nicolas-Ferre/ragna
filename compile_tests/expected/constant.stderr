error[E0308]: mismatched types
  --> src/constant.rs:8:9
   |
8  |         CONSTANT = 20;
   |         ^^^^^^^^
   |         |
   |         expected `&mut Gpu<i32, Mut>`, found `&mut Gpu<i32, Const>`
   |         arguments to this function are incorrect
   |
   = note: expected mutable reference `&mut Gpu<_, Mut>`
              found mutable reference `&mut Gpu<_, Const>`
note: method defined here
  --> ./crates/ragna/src/values.rs:80:12
   |
80 |     pub fn assign(&mut self, ctx: &mut GpuContext, value: Gpu<T, impl Any>) {
   |            ^^^^^^
